src/vs/workbench/services/editor/browser/editorResolverService.ts

```diff
@@ -42,6 +43,10 @@ type RegisteredEditors = Array<RegisteredEditor>;
 export class EditorResolverService extends Disposable implements IEditorResolverService {
 	readonly _serviceBrand: undefined;
 
+	// Events
+	private readonly _onDidChangeEditorRegistrations = this._register(new Emitter<void>());
```

Comment:
@lramos15 fyi I added this event to know when registrations come and go (also added a test)